rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -97,6 +97,16 @@ void SequentialWriter::open(
 
   const auto storage_uri = format_storage_uri(base_folder_, 0);
 
+  rcpputils::fs::path db_path(storage_uri);
+  if (!db_path.parent_path().is_directory()) {
```

Comment:
what happens if the directory exists already? 

I think the logic should be inverted here: There should be an exception being raised the moment the directory exists already. That's the current behavior in `ros2bag`. Simply because we don't want to overwrite any files.

Reply:
That makes sense, should be changed now.